app_dart/lib/src/model/appengine/task.dart

```diff
@@ -449,10 +449,12 @@ class Task extends Model<int> {
         return status = statusSucceeded;
       case Result.canceled:
         return status = statusCancelled;
-      case Result.infraFailure:
```

Comment:
Is infraFailure not needed? I would think since it is a separate ordinal it would still be viable to check for?

Reply:
It turns out that `Result` does not support `INFRA_FAILURE`, so we need to determine the infra failure case with the combination of `Result.failure` and `FailureReason.infraFailure`
See more details: https://github.com/flutter/flutter/issues/136908#issuecomment-1783273728

Reply:
Gotcha! Might be good to add a note here. Though this is likely to change with the migration to buildbucket v2 so up to you.

Reply:
Added a doc to make is clear. Also removed the existent `Result.infraFailure`.